src/fidesops/service/connectors/saas_connector.py

```diff
@@ -81,7 +65,11 @@ def get_authenticated_request(
             params=request_params.query_params,
             data=request_params.body,
         ).prepare()
-        return self.add_authentication(req, self.client_config.authentication)
+
+        authentication = self.client_config.authentication
+        return get_authentication_strategy(
+            authentication.strategy, authentication.configuration
+        ).add_authentication(req, self.secrets)
```

Comment:
It's a nit, but I think this reads more clearly as:

```suggestion
        auth_strategy = get_authentication_strategy(
            self.client_config.authentication.strategy,
            self.client_config.authentication.configuration,
        )
        
        return auth_strategy.add_authentication(req, self.secrets)
```

Overall though, I don't love that `add_authentication` is a method on `AuthenticationStrategy`, since it should really be on `PreparedRequest`. It just seems like strange usage, because you're adding authentication to the request, not the auth strategy. Obviously we can't make changes to `PreparedRequest` though, and I'm not seeing a better way to improve the usage 😕

Reply:
Changed this part so it's easier to read